libraries/AP_NavEKF2/AP_NavEKF2.cpp

```diff
@@ -614,66 +609,36 @@ const AP_Param::GroupInfo NavEKF2::var_info[] = {
 NavEKF2::NavEKF2()
 {
     AP_Param::setup_object_defaults(this, var_info);
-    _ahrs = &AP::ahrs();
-}
-
-/*
-  see if we should log some sensor data
- */
-void NavEKF2::check_log_write(void)
-{
-    if (!have_ekf_logging()) {
-        return;
-    }
-    if (logging.log_compass) {
-        AP::logger().Write_Compass(imuSampleTime_us);
-        logging.log_compass = false;
-    }
-    if (logging.log_baro) {
-        AP::logger().Write_Baro(imuSampleTime_us);
-        logging.log_baro = false;
-    }
-    if (logging.log_imu) {
-        AP::logger().Write_IMUDT(imuSampleTime_us, _logging_mask.get());
-        logging.log_imu = false;
-    }
-
-    // this is an example of an ad-hoc log in EKF
-    // AP::logger().Write("NKA", "TimeUS,X", "Qf", AP_HAL::micros64(), (double)2.4f);
 }
 
 
 // Initialise the filter
 bool NavEKF2::InitialiseFilter(void)
 {
+    AP::dal().start_frame(AP_DAL::FrameType::InitialiseFilterEKF2);
+
     // Return immediately if there is insufficient memory
     if (core_malloc_failed) {
         return false;
     }
 
     initFailure = InitFailures::UNKNOWN;
     if (_enable == 0) {
-        if (_ahrs->get_ekf_type() == 2) {
+        if (AP::dal().get_ekf_type() == 2) {
```

Comment:
We should probably create a `const auto &dal = AP::dal();` in here

Similarly in the EKF3 equivalent

Reply:
not worth it I think, and can't be a reference due to constructor ordering